common/src/main/java/org/opensearch/ml/common/output/model/MLResultDataType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.output.model;
+
+public enum MLResultDataType {
+    FLOAT32(Format.FLOATING, 4),
+    FLOAT64(Format.FLOATING, 8),
+    FLOAT16(Format.FLOATING, 2),
+    UINT8(Format.UINT, 1),
+    INT32(Format.INT, 4),
+    INT8(Format.INT, 1),
+    INT64(Format.INT, 8),
+    BOOLEAN(Format.BOOLEAN, 1),
+    UNKNOWN(Format.UNKNOWN, 0),
```

Comment:
Why do we need this `unknown` format. Will we be able to parse this type?

Reply:
There will be some gap between model output data type and java type. For such data types not recognized, we will set it as UNKNOWN.